File: Front_end/admin-app/src/features/order/domain/orderEvents.ts
export type OrderEventDefinition = {
  key: ORDER_EVENT_KEYS;
  label: string;
  description?: string;
};

export const ORDER_EVENTS: OrderEventDefinition[] = [
  {
    key: "order_created",
    label: "Order created",
    description: "Triggered when an order is created.",
  },
  {
    key: "order_confirmed",
    label: "Order confirmed",
    description: "Triggered when an order is confirmed.",
  },
  {
    key: "order_preparing",
    label: "Order preparing",
    description: "Triggered when an order is being prepared.",
  },
  {
    key: "order_ready",
    label: "Order ready",
    description:
      "Triggered when an order is ready to be delivered or picked up.",
  },
  {
    key: "order_processing",
    label: "Out for delivery",
    description: "Triggered when an order is out for delivery.",
  },
  {
    key: "order_completed",
    label: "Order completed",
    description: "Triggered when an order is completed.",
  },
  {
    key: "order_failed",
    label: "Order failed",
    description: "Triggered when an order fails.",
  },
  {
    key: "order_cancelled",
    label: "Order cancelled",
    description: "Triggered when an order is cancelled.",
  },
  {
    key: "order_delivery_plan_changed",
    label: "Order changed delivery plan",
    description: "Triggered when an order delivery plan changes.",
  },
  {
    key: "client_form_link_sent",
    label: "Client form link sent",
    description: "Triggered when a client form link is sent.",
  },
  {
    key: "order_rescheduled",
    label: "Order rescheduled",
    description:
      "Triggered when an order is rescheduled. It can trigger also if the order has state of 'Ready' and the arrival time changes.",
  },
];

export type ORDER_EVENT_KEYS =
  | "order_created"
  | "order_confirmed"
  | "order_preparing"
  | "order_ready"
  | "order_processing"
  | "order_completed"
  | "order_failed"
  | "order_cancelled"
  | "order_delivery_plan_changed"
  | "client_form_link_sent"
  | "order_rescheduled";

// DELIVERY_WINDOW_RESCHEDULED_BY_USER = "order_delivery_window_changed_by_user"

File: Back_end/Delivery_app_BK/services/domain/order/order_events.py
from enum import Enum


class OrderEvent(str, Enum):
    CREATED = "order_created"
    CONFIRMED = "order_confirmed"
    PREPARING = "order_preparing"
    EDITED = "order_edited"
    READY = "order_ready"
    PROCESSING = "order_processing"
    COMPLETED = "order_completed"
    FAIL = "order_failed"
    CANCELLED = "order_cancelled"
    STATUS_CHANGED = "order_status_changed"
    DELIVERY_WINDOW_RESCHEDULED_BY_USER = "order_delivery_window_changed_by_user"
    DELIVERY_PLAN_CHANGED = "order_delivery_plan_changed"
    DELIVERY_RESCHEDULED = "order_rescheduled"
    CLIENT_FORM_LINK_SENT = "client_form_link_sent"


class OrderEventPrintDocuments(str,Enum):
    CREATED = OrderEvent.CREATED.value
    CONFIRMED = OrderEvent.CONFIRMED.value
    EDITED = OrderEvent.EDITED.value


"""
Delivery plan changed happes when a order changes of delivery plan type, 

Order_rescheduled happens when a delivery plan changes it's dates. or when an order changes
of plan and dates are not the same as the previous plan dates 

DELIVERY_WINDOW_RESCHEDULED_BY_USER happens when a user changes the window of an order manually,
this windows are bounderies not clear dates 


"""
